main/integration-tests/support/resources/base/resource.js

```diff
@@ -42,6 +42,28 @@ class Resource {
     }
   }
 
+  // When creating this resource, this method provides default values.
+  // Extender should override this method and implement their own logic for providing default values
+  defaults(resource = {}) {
+    return resource;
+  }
+
+  async create(body = {}, params = {}, { api = this.api, applyDefault = true } = {}) {
+    try {
+      const requestBody = applyDefault ? this.defaults(body) : body;
+      const response = await this.axiosClient.post(api, requestBody, { params });
+      const resource = response.data;
+      const taskId = `${this.type}-${this.id}`;
+
```

Comment:
What if this resource itself is a parent of a collection resource? Would we not need the `childIdProp` for it?

Reply:
Reasoning discussed separately.